op-tests/test-bls-ops.txt

```diff
@@ -0,0 +1,209 @@
+coinid 0x0000000000000000000000000000000000000000000000000000000000000000 0x0000000000000000000000000000000000000000000000000000000000000000 123456789 => 0x2f6c01d9205e70f55b940367dcdc1b518f077ee7308788ec4467447708a76e79 | 633
+coinid 0x1234500000000000000000000000000000000000000000000000000000000000 0x6789abcdef000000000000000000000000000000000000000000000000000000 123456789 => 0x69bfe81b052bfc6bd7f3fb9167fec61793175b897c16a35827f947d5cc98e4bc | 633
+coinid 0x1234500000000000000000000000000000000000000000000000000000000000 0x6789abcdef000000000000000000000000000000000000000000000000000000 0x00ffffffffffffffff => 0x609d2d5e3081fbc1106950950f3ea3dbb4eaec96a57a544ba83b8a762b457168 | 633
+
+; parent coin ID has the wrong size
+coinid 0x123450000000000000000000000000000000000000000000000000000000000000 0x6789abcdef000000000000000000000000000000000000000000000000000000 123456789 => FAIL
+coinid 0x12345000000000000000000000000000000000000000000000000000000000 0x6789abcdef000000000000000000000000000000000000000000000000000000 123456789 => FAIL
+
+; puzzle hash has the wrong size
+coinid 0x1234500000000000000000000000000000000000000000000000000000000000 0x6789abcdef00000000000000000000000000000000000000000000000000000000 123456789 => FAIL
+coinid 0x1234500000000000000000000000000000000000000000000000000000000000 0x6789abcdef0000000000000000000000000000000000000000000000000000 123456789 => FAIL
+
+; amount is invalid
+coinid 0x1234500000000000000000000000000000000000000000000000000000000000 0x6789abcdef000000000000000000000000000000000000000000000000000000 0x010000000000000000 => FAIL
+coinid 0x1234500000000000000000000000000000000000000000000000000000000000 0x6789abcdef000000000000000000000000000000000000000000000000000000 -1 => FAIL
+coinid 0x1234500000000000000000000000000000000000000000000000000000000000 0x6789abcdef000000000000000000000000000000000000000000000000000000 0x00001234 => FAIL
+
+; bls g1 subtraction
+; expects (G1point G1point ...)
+bls_g1_subtract 0x97f1d3a73197d7942695638c4fa9ac0fc3688c4f9774b905a14e3a3f171bac586c55e83ff97a1aeffb3af00adb22c6bb 0xa572cbea904d67468808c8eb50a9450c9721db309128012543902d0ac358a62ae28f75bb8f1c7c42c39a8c5529bf0f4e => 0xb7f1d3a73197d7942695638c4fa9ac0fc3688c4f9774b905a14e3a3f171bac586c55e83ff97a1aeffb3af00adb22c6bb | 2857918
+; identity
+bls_g1_subtract => 0xc00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000 | 132812
+; G1 point has invalid length
+bls_g1_subtract 0x97f1d3a73197d7942695638c4fa9ac0fc3688c4f9774b905a14e3a3f171bac586c55e83ff97a1aeffb3af00adb22c6 => FAIL
+bls_g1_subtract 0x97f1d3a73197d7942695638c4fa9ac0fc3688c4f9774b905a14e3a3f171bac586c55e83ff97a1aeffb3af00adb22c6bb00 => FAIL
+bls_g1_subtract 0 => FAIL
+bls_g1_subtract ( 1 2 3 ) => FAIL
```

Comment:
Does only one argument work?  If not, these tests could be failing for that reason and not for the length reasons right?  If one argument is allowed, why?  I would think subtraction requires at least 2 (maybe maximum, like I believe the arithmetic subtract works).

Reply:
(I was wrong about the fixed number of args for subtraction so ignore the parenthesis)

Reply:
I will make sure we have all those cases covered by tests though